### DISCUSSION TITLE
Help with query filter

### DISCUSSION THREAD

**Discussion by jgosnell56** [bubuntux/nordlynx#25] [Q&A]
Hi All, I see that most people are using the following filter to specify a country to connect to: 
- QUERY=filters\[country_id\]=227

I want to connect to a server type rather than a county.  I am trying to use the filter below but do not think it is working right, can someone help me with the correct syntax to only connect to P2P servers no matter the country?
![filter](https://user-images.githubusercontent.com/67598418/151707536-8d91befa-44f9-417a-963b-66021c0897e2.png)

EDIT 02/03/2022:
I believe the filter I have is actually working to parse only P2P servers but I cannot combine with a country filter like this:
`filters\[country_id]=228\&filters\[servers_groups\]\[identifier\]=legacy_p2p`

Wireguard will not start in the container and I see a config parsing error.  Does that query need to be modified somehow to make it work in the container?  It works directly in a web page to query the API.

**Comment by nrk666** [comment]
Is it even possible to stack filters on the QUERY env var? I'm interested in this as well. 

Example: 

I'd like to connect to P2P servers in New York. Does this work : 

QUERY=filters\\[city_id\\]=8971718,\\[servers_groups\\]\\[identifier\\]=legacy_p2p 

and if not, if possible, what would be the correct syntax?

**Reply by jensbodal** [reply]
does this work for you?

**Comment by ThxAndBye** [comment]
You need to use the URL syntax of the NordVPN API but need to escape for `curl` used [here](https://github.com/bubuntux/nordlynx/blob/032a697890ffe9d488fd0aa4eca997e998790df6/root/etc/services.d/wireguard/run#L8).  
To use multiple, you'd use it like this: `filters\[country_id\]=209&filters\[servers_groups\]\[identifier\]=legacy_p2p`  
Important is to escape (`\`) only characters that you actually want/need to escape. The `&` must not be escaped.

**Reply by nrk666** [reply]
perfect, that was exactly what I needed, thanks!

**Reply by imthenachoman** [reply]
What characters need to be escaped?

**Comment by b-kamphorst** [comment]
Hi all,

Thanks to @ThxAndBye I got everything seemingly working fine. However, I'm not sure whether the connected server actually meets the criteria from my query. In particular, my public IP does not pop up in the query results.

My current query is the following:
`QUERY=filters\[country_id\]=153&filters\[servers_groups\]\[identifier\]=legacy_p2p`

In the nordlynx container, my public IP is listed as
```bash
$ curl ipinfo.io/ip
176.119.195.136
```
This IP seems to be from a NordVPN server in Amsterdam. All seems fine so far.

However, a manual query ([like nordlynx](https://api.nordvpn.com/v1/servers/recommendations?&filters%5C[servers_technologies%5C]%5C[identifier%5C]=wireguard_udp&limit=1&filters%5C[country_id%5C]=153&filters%5C[servers_groups%5C]%5C[identifier%5C]=legacy_p2p) with limit=1, [without limit](https://api.nordvpn.com/v1/servers/recommendations?&filters%5C[servers_technologies%5C]%5C[identifier%5C]=wireguard_udp&filters%5C[country_id%5C]=153&filters%5C[servers_groups%5C]%5C[identifier%5C]=legacy_p2p)) doesn't show that IP anywhere. This confuses me. 

Does someone have an idea to what is happening here? Does it make sense, or is this an issue?
